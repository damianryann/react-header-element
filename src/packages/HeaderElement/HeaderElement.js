import React, { Fragment } from 'react';

function HeaderElement(props) {
  const { title, isFunctional, value, fallBack, className, id, style } = props;

  const HeaderElement = value ? value : fallBack || 'h1';

  function addClassToDiv() {
    let classes = [HeaderElement];

    if (className) {
      classes.push(className);
    }

    return classes.join(' ');
  }

  function addClassToTag() {
    let tagClasses = (document.getElementsByTagName(
      HeaderElement
    ).className = className);

    return tagClasses;
  }

  function addId() {
    let tagIdToTag = (document.getElementsByTagName(HeaderElement).id = id);
    let tagIdToClass = (document.getElementsByClassName(HeaderElement).id = id);

    return tagIdToTag || tagIdToClass;
  }

  function addStyle() {
    let tagIdToTag = (document.getElementsByTagName(
      HeaderElement
    ).style = style);
    let tagIdToClass = (document.getElementsByClassName(
      HeaderElement
    ).style = style);

    return tagIdToTag || tagIdToClass;
  }

  return (
    <Fragment>
      {isFunctional === true ? (
        <HeaderElement
          className={addClassToTag()}
          id={addId()}
          style={addStyle()}
        >
          {title}
        </HeaderElement>
      ) : (
        <div className={addClassToDiv()} id={addId()} style={addStyle()}>
          {title}
        </div>
      )}
    </Fragment>
  );
}

export default HeaderElement;
