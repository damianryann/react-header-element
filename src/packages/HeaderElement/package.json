{
    "name": "react-header-element",
    "version": "1.0.2",
    "main": "HeaderElement.js",
    "author": "Damian Ryan <damian@damianryan.co.uk>",
    "license": "MIT"
}