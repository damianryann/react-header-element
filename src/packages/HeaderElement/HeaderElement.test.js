import React from 'react';
import { mount } from 'enzyme';
import HeaderElement from './HeaderElement';

import testData from './HeaderElement.test.json';

it('mounts without crashing', () => {
  const component = mount(<HeaderElement />);

  expect(component).toMatchSnapshot();
});

it('is isFunctional true', () => {
  const component = mount(<HeaderElement {...testData[0]} />);

  const { isFunctional } = testData;

  if (isFunctional) {
    expect(component.find('h1')).toHaveLength(1);
    expect(isFunctional).toBeTruthy();
  }

  expect(component).toMatchSnapshot();
});

it('is isFunctional false', () => {
  const component = mount(<HeaderElement {...testData[1]} />);

  const { isFunctional } = testData;

  if (isFunctional) {
    expect(component.find('h2')).toHaveLength(1);
    expect(isFunctional).toBeFalsy();
  }

  expect(component).toMatchSnapshot();
});

it('is value an h2 and is fallBack h1', () => {
  const component = mount(<HeaderElement {...testData[1]} />);

  const { value, fallBack } = testData;

  expect(component.find('HeaderElement')).toHaveLength(1);

  if (value) {
    expect(value).toBe('h2');
    expect(fallBack).toBe('h1');
  }

  expect(component).toMatchSnapshot();
});
