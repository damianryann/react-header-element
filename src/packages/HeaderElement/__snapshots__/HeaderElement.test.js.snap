// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is isFunctional false 1`] = `
<HeaderElement
  className="another-class"
  fallBack="h1"
  id="another-id"
  isFunctional={false}
  title="This is a Header"
  value="h2"
>
  <div
    className="h2 another-class"
    id="another-id"
  >
    This is a Header
  </div>
</HeaderElement>
`;

exports[`is isFunctional true 1`] = `
<HeaderElement
  className="class-name"
  fallBack="h2"
  id="id-name"
  isFunctional={true}
  title="This is a Header"
  value="h1"
>
  <h1
    className="class-name"
    id="id-name"
  >
    This is a Header
  </h1>
</HeaderElement>
`;

exports[`is value an h2 1`] = `
<HeaderElement
  className="another-class"
  fallBack="h1"
  id="another-id"
  isFunctional={false}
  title="This is a Header"
  value="h2"
>
  <div
    className="h2 another-class"
    id="another-id"
  >
    This is a Header
  </div>
</HeaderElement>
`;

exports[`is value an h2 and is fallBack h1 1`] = `
<HeaderElement
  className="another-class"
  fallBack="h1"
  id="another-id"
  isFunctional={false}
  title="This is a Header"
  value="h2"
>
  <div
    className="h2 another-class"
    id="another-id"
  >
    This is a Header
  </div>
</HeaderElement>
`;

exports[`is value an h2 and is fallBack h1 2`] = `
<HeaderElement
  className="another-class"
  fallBack="h1"
  id="another-id"
  isFunctional={false}
  title="This is a Header"
  value="h2"
>
  <div
    className="h2 another-class"
    id="another-id"
  >
    This is a Header
  </div>
</HeaderElement>
`;

exports[`mounts without crashing 1`] = `
<HeaderElement>
  <div
    className="h1"
  />
</HeaderElement>
`;
